atomic_reactor/__init__.py

```diff
@@ -32,6 +35,28 @@ def format(self, record):
         return super(ArchFormatter, self).format(record)
 
 
+class EncodedStream(object):
+    def __init__(self, fileno, encoding):
+        self.binarystream = fdopen(dup(fileno), 'wb')
```

Comment:
There's probably a very good reason why the file descriptor is being duplicated here, but won't this cause overlapping writing issues?